auto_nag/scripts/regression_without_regressed_by.py

```diff
@@ -0,0 +1,180 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from libmozdata.bugzilla import Bugzilla, BugzillaUser
+from auto_nag.bzcleaner import BzCleaner
+from auto_nag import utils
+
+
+class RegressionWithoutRegressedBy(BzCleaner):
+    def __init__(self):
+        super(RegressionWithoutRegressedBy, self).__init__()
+
+    def description(self):
+        return 'Regressions without regressed_by and some dependencies'
+
+    def handle_bug(self, bug, data):
+        bugid = bug['id']
+        deps = set(bug['blocks']) | set(bug['depends_on'])
+        deps = set(x for x in deps if x < bugid)
```

Comment:
Maybe instead of checking the bug ID, we could check if the regression bug was opened after the possible regressor bug was closed.

By checking the bug ID we only ensure that the regression bug was opened after the possible regressor bug was closed, so we would remove fewer bugs than if we checked the resolved date.

Reply:
(I'd also move all the filtering in the same place, so we can see it all together)

Reply:
So we'd need to get the history of the dependency bug and see what was its state when the current bug has been opened => need to get history: it seems to me a quite overkill.
Did you see some bugs where it was inaccurate to ask to fill regressed_by and a dependency not closed before the regression ?

About the other point: (i.e. move filtering...)
A part of the filtering is made in handle_bug (remove bugid > current from dependencies) and the other part is made in filter_bugs (remove meta or never resolved bugs).
The second step requires new bugzilla query to get info about bug dependencies, so removing some useless data in the first step allow us to reduce queries in the second step.

Reply:
> So we'd need to get the history of the dependency bug and see what was its state when the current bug has been opened => need to get history: it seems to me a quite overkill.
Did you see some bugs where it was inaccurate to ask to fill regressed_by and a dependency not closed before the regression ?
> Did you see some bugs where it was inaccurate to ask to fill regressed_by and a dependency not closed before the regression ?

I haven't checked whether it happened in your current results, but in theory it could happen. The less wrong results we have the better.
It shouldn't be too overkill, it's really a small number of bugs, so it should be pretty quick to retrieve their data. We can do it as a follow-up.

Reply:
> About the other point: (i.e. move filtering...)
A part of the filtering is made in handle_bug (remove bugid > current from dependencies) and the other part is made in filter_bugs (remove meta or never resolved bugs).
The second step requires new bugzilla query to get info about bug dependencies, so removing some useless data in the first step allow us to reduce queries in the second step.

Couldn't you do the first part of the filtering in filter_bugs, before making the new bugzilla query?